### DISCUSSION TITLE
[BUG] Rule Keys are are different in Log Explorer and rulesets causing no alerts

### DISCUSSION THREAD

**Discussion by thedunston** [utmstack/UTMStack#820] [Q&A]
**Describe the bug**

I am using UTMStack 10.5.6. I got it setup in a VM and the install went smoothly. I was able to login, I don't see any errors. However, I noticed that there were no alerts being generated. I reviewed the rules and tried generating events and nothing happened.

I then noticed that the key of the rules in the "Log Explorer" view and the "Manage Correlation rules" have a few differences. The rules below:

logx.wineventlog.event_data.ParentProcessName  => logx.wineventlog.event_data.ParentImage
logx.wineventlog.event_data.ProcessName  => logx.wineventlog.event_data.Image

are of interest.  On the left, is what is in the `rules` for windows events, but on the right is what the `Log Explorer` is mapping the log key too.  Once I updated all the windows based events, then I started getting alerts.  Namely, the `ParentProcessName` and `ProcessName` are being logged as `ParentImage` and `Image`, respectively.

I also see that the logs are very similar to Sigma Rules. Can you all create a parser for Sigma rules to the UTMStack format or use the Sigma Rule format?

**To Reproduce**

1. Go to '...'  `Log Explorer` and search for an event in `logx.wineventlog`.
2. Click on '....' `Manage correlation rules`
3. Scroll down to '....' =>  `System => windows` and open a rule.  You'll notice the `Log Explorer` key is different than the rule for `ProcessName` and `ParentProcessName`.

**Possible solution**

he rules need to be updated to reflect that change for  `logx.wineventlog`.

**Comment by osmontero** [comment]
@c3s4rfred please check it.

**Comment by osmontero** [comment]
@thedunston please provide more information/evidence related to the issue (screenshots, sample logs, ...).

**Comment by thedunston** [comment]
Please see the attached screenshots and reposting steps:
To Reproduce

Steps to reproduce the behavior:

Go to '...' Log Explorer and search for an event in logx.wineventlog.
Click on '....' Manage correlation rules
Scroll down to '....' => System => windows and open a rule. You'll notice the Log Explorer key is different than the rule for ProcessName and ParentProcessName.
![LogExplorerView](https://github.com/user-attachments/assets/a1c92d8d-9e31-4f38-90cf-93bcb66ba7a0)
![RuleThatComesWithUTMStack](https://github.com/user-attachments/assets/443cacc7-d510-4369-afce-825643cab044)
![alertViewUpClose](https://github.com/user-attachments/assets/bc928203-c72c-419c-b8c9-ad09da22d2b1)
![RuleAfterChangingFromProcessNameToImage](https://github.com/user-attachments/assets/41d14d11-819c-423c-bb17-3cba6a17dec9)
![alertView](https://github.com/user-attachments/assets/6d79054f-aad7-4357-9d8f-fdafce328f37)

**Comment by thedunston** [comment]
Any updates on this issue? Alerting is critical.

**Comment by osmontero** [comment]
That rule was created based on example logs we got in our lab. And the field names in the example logs are the ones we included in the rule. If for some reason in your instance you are receiving data in different fields you can make a copy of the rule and add it in an additional folder outside of the system rules folder so that it is not removed during upgrades. I will move this to the discussions as we do not identify this as a bug.

**Comment by thedunston** [comment]
You all wrote code to pull the latest rules from the GitHub repo:

https://github.com/utmstack/UTMStack/blob/main/correlation/rules/update.go

which also contains the `.ProcessName`. Why have this in the production versions of UTMStack if it is a test you all do in the lab that does not work? This issue exists on the hosted Cloud Platform as well. My plan was to evaluate and then purchase a Cloud License. However, I don't understand why you all advertise having all these rules available and then I have to modify the rules to make it work.

**Reply by osmontero** [reply]
In our tests the event is received with the fields that were included in the rule and not in the fields you mention. Since UTMStack allows you to create custom rules, and in your case, your Windows installation seems to be reporting that information in an unexpected field for some reason, you can create your own rule evaluating that field instead of the one we normally receive in our tests.
https://github.com/utmstack/rules.git is the official UTMStack rules repository, so it's fine to download them from there.
